R/shiny.R

```diff
@@ -622,19 +654,46 @@ sleuth_live <- function(obj, ...) {
     })
 
     ### DE table
+    output$no_gene_error <- renderUI({
+        if(input$pop_genes == 2) {
+            wb <- input$which_beta_de
+            if ( is.null(wb) ) {
+                poss_tests <- tests(models(obj, verbose = FALSE)[[input$which_model_de]])
+                wb <- poss_tests[1]
+            }
+            if(!('ens_gene' %in% colnames(sleuth_results(obj, wb, input$which_model_de)))) {
+                HTML('You need to add genes to your sleuth object before you can view a popped gene table.<br> See the <a href="http://pachterlab.github.io/sleuth/starting.html">sleuth getting started guide</a> for how to add genes to a sleuth object.')
+            }
+        }
+    })
+    
     output$which_beta_ctrl_de <- renderUI({
       poss_tests <- tests(models(obj, verbose = FALSE)[[input$which_model_de]])
       selectInput('which_beta_de', 'beta: ', choices = poss_tests)
     })
-
+    
     output$de_dt <- renderDataTable({
       wb <- input$which_beta_de
       if ( is.null(wb) ) {
         poss_tests <- tests(models(obj, verbose = FALSE)[[input$which_model_de]])
         wb <- poss_tests[1]
       }
-
-      sleuth_results(obj, wb, input$which_model_de)
+      res = sleuth_results(obj, wb, input$which_model_de)
+      
+      if(input$pop_genes == 2) {
+          if('ens_gene' %in% colnames(res)) {
+              popped_gene_table = res
```

Comment:
can you factor our the `popped_gene_table` into a function in `sleuth.R`. let's call it... `sleuth_gene_table` for now